Src/main.c

```diff
@@ -99,15 +100,27 @@ int main(void)
   /* USER CODE BEGIN 2 */
   debug("\n\nBootcamp starting up...");
   debug("Compiled on %s at %s", __DATE__, __TIME__);
-
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Start(&htim16,TIM_CHANNEL_1);
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  int minDutyCycleMs = 1;
+  int maxDutyCycleMs = 2;
+  int differenceDutyCycleMs = maxDutyCycleMs - minDutyCycleMs;
+  int arrPeriod = 60000; //arrPeriod in this context is equal to the maxDutyCycleMs
+
   while (1)
   {
+    uint16_t adcOutput = 0;
+    adcOutput = HAL_ADC_GetValue(&hadc); //This will give a value corrosponding to 0-4095 and will be between 0-3.3v. It is a 12 bit long int.
+    double adcPercentage = adcOutput/4095;
```

Comment:
Guaranteed you'll never make this mistake again in your life LOL. 

I ran these couple lines in an online c++ compiler, to see if they do the right thing. As it turns out, if adcOutput is anything less than 4095, then adcPercentage evaluates to 0. 

Here's what's going on. The compiler encounters the division " adcOutput/4095". Since adcOutput is an integer, and so is the literal "4095", integer division gets performed (no decimal points) and the result is 0. But since that value needs to be assigned to a double, the compiler implicitly converts it to a double. So adcPercentage gets assigned 0.000000000000000.

What you really want to do is get the compiler to perform some sort of double division. To do that, as per C's "Usual arithmetic conversions" (https://docs.microsoft.com/en-us/cpp/c-language/usual-arithmetic-conversions?view=vs-2019) (a highly recommended quick read), you'll need at least one of the operands to be a double.

I'll let you figure out how to make the litteral "4095" a double and how to cast adcOutput to a double.


A quick side note: This bug is a great example of the sort of stuff unit testing is meant to catch. It would be hard to trace back this problem once things were running on the actual aircraft, but very easy to find the issue with a simple test case.